botocore/auth.py

```diff
@@ -344,14 +341,16 @@ def _set_necessary_date_headers(self, request):
         # header.  Otherwise we use the X-Amz-Date header.
         if 'Date' in request.headers:
             del request.headers['Date']
+            datetime_timestamp = datetime.datetime.strptime(
+                request.context['timestamp'], '%Y%m%dT%H%M%SZ')
```

Comment:
Any reason to use `strptime` here on the string instead of just storing the datetime object in the context alongside the string? This seems like extra work without any real benefit unless I'm mistaken.